client/app/src/main/java/group01/smartcar/client/MainActivity.java

```diff
@@ -1,63 +1,126 @@
-import static group01.smartcar.client.Status.*;
 
-public class MainActivity extends AppCompatActivity implements JoystickView.JoystickListener{
-    CarControl car;
-    protected ImageView cameraView;
+import com.google.firebase.auth.FirebaseAuth;
+import com.google.firebase.auth.FirebaseUser;
+
+public class MainActivity extends AppCompatActivity {
+    private FirebaseAuth firebaseAuth;
+    EditText emailTextView, passwordTextView;
+    private VideoView videoBackground;
+    MediaPlayer mediaPlayer;
+    int currentVideoPosition;
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_main);
-        registerComponentCallbacks();
-        cameraView = findViewById(R.id.imageView);
-        car = new CarControl(this.getApplicationContext(), cameraView);
-    }
 
-    @Override
-    protected void onPause() {
-        super.onPause();
-        car.pause();
-    }
+        ActionBar actionBar = getSupportActionBar();
+        assert actionBar != null;
+        actionBar.hide();
 
-    @Override
-    protected void onResume() {
-        super.onResume();
-        System.out.println("OnResume");
-        // Reconnect to MQTT server if application is resumed
-        car.resume();
+        videoBackground = findViewById(R.id.videoView);
+        videoBackground.getHolder().setSizeFromLayout();
+        loadBackground();
+
+        firebaseAuth = FirebaseAuth.getInstance();
+        registerComponentCallbacks();
     }
 
     private void registerComponentCallbacks() {
-        findViewById(R.id.start_button).setOnClickListener(this::onStartClick);
-        findViewById(R.id.stop_button).setOnClickListener(this::onStopClick);
+        emailTextView = findViewById(R.id.email_textfield);
+        passwordTextView = findViewById(R.id.password_textfield);
+        findViewById(R.id.login_button).setOnClickListener(login);
     }
 
-    private void onStartClick(View view) {
-        car.start();
+    private final View.OnClickListener login = new View.OnClickListener() {
+        @Override
+        public void onClick(View v) {
+            String email = emailTextView.getText().toString();
+            String password = passwordTextView.getText().toString();
+
+            firebaseAuth.signInWithEmailAndPassword(email, password).addOnCompleteListener(task -> {
+                if (task.isSuccessful()) {
+                    Log.d("EmailPassword", "signInWithEmail:success");
+                    verifyUser();
+                } else {
+                    Log.w("EmailPassword", "signInWithEmail:failure",
+                            task.getException());
+                    Toast.makeText(
+                            MainActivity.this, "Failed to log in! Ask your " +
+                                    "local AlSet dealer for your " +
+                                    "personal login details.", Toast.LENGTH_SHORT).show();
```

Comment:
I found it hard seeing the error message.
I suggest we center it above the keyboard for better visibility.

![ToastMessage](https://user-images.githubusercontent.com/70911363/116043779-61113e00-a670-11eb-8bb2-87c2b4f47f01.png)

Reply:
Hmm, many people are used to seeing their toasts at the bottom of the screen. It sounds a lot more natural to put it there, so I'm not sure where you'd prefer this Toast message to be displayed. Additionally, the error message is there temporarily and it can be assumed that you failed to log in if you weren't able to go past the login screen.

I'm also not sure with the visibility issue, I can see it just fine. Thoughts?

Reply:
Maybe it's just me feeling it's weird using it from the emulator. :)
Since it's default position is there it might be my eyesight that needs some changes.